@here/harp-texture-atlas-generator/src/ImageBitmapCoders.ts

```diff
@@ -0,0 +1,212 @@
+ * Supported formats are: JPEG, PNG, BMP, TIFF, GIF.
+ */
+class ImageBitmapEncoder implements ImageEncoder {
+    readonly width: number;
+    readonly height: number;
+    private readonly m_image: Jimp.Jimp;
+
+    constructor(image: Jimp.Jimp) {
+        this.m_image = image;
+        this.width = image.bitmap.width;
+        this.height = image.bitmap.height;
+    }
+
+    setPixelAt(x: number, y: number, color: Color): void {
+        if (x < this.width && y < this.height) {
+            try {
+                this.m_image.setPixelColor(
+                    Jimp.rgbaToInt(color.r, color.g, color.b, color.a), x, y,
+                    (err: Error|null, image: Jimp) => {
+                        if (err) {
+                            throw new Error(`Can not convert pixel value:
+                                ${JSON.stringify(color)}!`);
+                        }
+                    });
+            }
+            catch (err) {
+                // Jimp does not throw standarized errors so we need to handle it here.
+                throw new Error(`Can not write pixel value:
+                    ${JSON.stringify(color)}!`);
+            }
+        }
+        else {
+            throw new Error(`Access to pixel outside the image boundary!`);
+        }
+    }
+
+    getPixelAt(x: number, y: number): Color {
+        if (x < this.width && y < this.height) {
+            return getJimpPixelAt(this.m_image, x, y);
+        }
+        else {
+            throw new Error(`Access to pixel outside the image boundary!`);
+        }
+    }
+
+    copy(): ImageEncoder {
+        return ImageBitmapEncoderConstructor.clone(this.m_image);
+    }
+
+    resize(width: number, height: number): ImageEncoder {
+        return new ImageBitmapEncoder(
+            this.m_image.clone().resize(width, height, Jimp.RESIZE_BICUBIC));
+    }
+
+    /**
+     * Write image to file asynchronously.
+     * The image can be written to disk in PNG, JPEG or BMP format (based on the file extension)
+     * or if no extension is provided the original image's MIME type which, if not available,
+     * defaults to PNG).
+     *
+     * @param filePath file storage path.
+     * @returns Promise.
+     */
+    write(filePath: string): Promise<any> {
+        return new Promise((resolve, reject) => {
+            if (!checkExportPath(filePath)) {
+                reject(new Error("Failed to write and image: " + filePath +
+                    " only BMP, PNG, JPG formats are supported"));
+            }
+            this.m_image.write(filePath, (err, image) => {
+                if (err) {
+                    reject(new Error("Failed to write and image: " + filePath));
+                }
+                else {
+                    resolve();
+                }
+            });
+        });
+    }
+
+    /**
+     * Write image to disk synchronously.
+     * @see write.
+     *
+     * @param filePath file storage path.
+     */
+    writeSync(filePath: string): void {
```

Comment:
Should be removed because it's impossible to implment in most cases (commented later).

Quick search says: it's not used at all.